scrapy/utils/misc.py

```diff
@@ -160,6 +160,46 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
     return instance
 
 
+def build_from_crawler(objcls, crawler, /, *args, **kwargs):
+    """Construct a class instance using its ``from_crawler`` constructor.
+
+    ``*args`` and ``**kwargs`` are forwarded to the constructor.
+
+    Raises ``TypeError`` if the resulting instance is ``None``.
+    """
+    settings = crawler.settings
+    if crawler and hasattr(objcls, "from_crawler"):
```

Comment:
I think we shouldn't check `if crawler` and assume it's not None if this method was called.